mslice/plotting/plot_window/cut_plot.py

```diff
@@ -150,20 +150,29 @@ def get_line_options_by_index(self, line_index):
         line_options['style'] = line.get_linestyle()
         line_options['width'] = str(int(line.get_linewidth()))
         line_options['marker'] = line.get_marker()
+        line_options['error_bar'] = self._single_line_has_error_bars(line_index)
         return line_options
 
     def set_line_options_by_index(self, line_index, line_options):
         container = self._canvas.figure.gca().containers[line_index]
         container.set_label(line_options['label'])
-        main_line = container.get_children()[0]
+        main_line, error_bar_part = container.get_children()[0], container.get_children()[1:]
         main_line.set_linestyle(line_options['style'])
         main_line.set_marker(line_options['marker'])
+
         self._legends_visible[line_index] = bool(line_options['legend'])
         for child in container.get_children():
             child.set_color(line_options['color'])
             child.set_linewidth(line_options['width'])
             child.set_visible(line_options['shown'])
 
+        if not line_options['error_bar'] and self.get_line_visible(line_index):
+            for i in error_bar_part:
+                i.set_alpha(0)
+        else:
+            for i in error_bar_part:
+                i.set_alpha(1)
```

Comment:
This can be collapsed into a single if and for loop:
```
        if line_options['shown']:
            for i in error_bar_part:
                i.set_alpha(line_options['error_bar'])
```